test/query_test.js

```diff
@@ -312,6 +324,136 @@ describe('query', function () {
     });
   });
 
+  // Type check functions
+
+  it ('type check', async function() {
+    const coll = await serverClient.query(
+      query.CreateCollection({ name: util.randomString('collection_') })
+    ).then(pluckRef);
+    const index = await serverClient.query(query.CreateIndex({
+      name: util.randomString('index_'),
+      source: coll,
+      active: true,
+    })).then(pluckRef);
+    const doc = await serverClient.query(
+      query.Create(coll, { credentials: { password: 'sekret' } })
+    ).then(pluckRef);
+    const db = await adminClient.query(
+      query.CreateDatabase({ name: util.randomString('db_') })
+    ).then(pluckRef);
+    const fn = await serverClient.query(query.CreateFunction({
+      name: util.randomString('fn_'),
+      body: query.Query(query.Lambda('x', query.Var('x'))),
+     })).then(pluckRef);
+    const key = await adminClient.query(
+      query.CreateKey({ database: db, role: 'admin' })
+    ).then(pluckRef);
+    const tok = await serverClient.query(query.Login(doc, { password: 'sekret' }));
+    const role = await adminClient.query(query.CreateRole({
+      name: util.randomString('role_'),
+      membership: [],
+      privileges: [],
+    })).then(pluckRef);
+    // const sessionClient = new Client({ ...util.getCfg(), secret: tok.secret });
+    // const cred = await sessionClient.query(query.Paginate(query.Credentials()));
```

Comment:
jvm:
```
    FaunaClient cli = serverClient.newSessionClient(tok.get(SECRET_FIELD));
    Value cred = cli.query(Get(Ref("credentials/self"))).get().at("ref");
```
i guess this is not the correct equivalent. any suggestions @marrony ?

Reply:
no this is not, the only way to get credential is using the old ref model `Ref("credentials/self")`

Reply:
Thanks. I had tried this previously without success, but it's working now.